underfs/s3a/src/main/java/alluxio/underfs/s3a/S3AInputStream.java

```diff
@@ -123,56 +131,40 @@ public long skip(long n) throws IOException {
     if (n <= 0) {
       return 0;
     }
-    closeStream();
-    mPos += n;
-    openStream();
-    return n;
+    long skipped = in.skip(n);
+    mPos += skipped;
+    return skipped;
   }
 
-  /**
-   * Opens a new stream at mPos if the wrapped stream mIn is null.
-   */
-  private void openStream() throws IOException {
-    if (mIn != null) { // stream is already open
+  @Override
+  public long getPos() {
+    return mPos;
+  }
+
+  @Override
+  public void seek(long pos) throws IOException {
+    if (pos == mPos) {
       return;
     }
-    GetObjectRequest getReq = new GetObjectRequest(mBucketName, mKey);
-    // If the position is 0, setting range is redundant and causes an error if the file is 0 length
-    if (mPos > 0) {
-      getReq.setRange(mPos);
+    if (pos < mPos) {
+      mPos = pos;
+      reopenStream();
+      return;
     }
-    AmazonS3Exception lastException = null;
-    String errorMessage = String.format("Failed to open key: %s bucket: %s, left retry:%d",
-        mKey, mBucketName, mRetryPolicy.getAttemptCount());
-    while (mRetryPolicy.attempt()) {
-      try {
-        mIn = getClient().getObject(getReq).getObjectContent();
-        return;
-      } catch (AmazonS3Exception e) {
-        errorMessage = String
-            .format("Failed to open key: %s bucket: %s attempts: %d error: %s", mKey, mBucketName,
-                mRetryPolicy.getAttemptCount(), e.getMessage());
-        throw AlluxioS3Exception.from(errorMessage, e);
```

Comment:
@jja725  this one is similar to before, so we decide to roll back to still use IOException?

Reply:
Previously we use IOException (e.g. 2.7 time), but in 2.8 and 2.9 it's AmazonS3Exception, do we want to rollback?